src/mount.js

```diff
@@ -0,0 +1,42 @@
+const ensureFunction = require('type/function/ensure')
+const ensureValue = require('type/value/ensure')
+const { startWatcher } = require('./mutation-observer')
+const { registerHtml } = require('./dom-wrappers')
+const { TRAM_MUTATION_OBSERVER } = require('./engine-names')
+
+/**
+ * @private
+ * @description
+ * Updates a selector with an initial component for the first render.
+ */
+module.exports = (selector, component) => {
+	ensureValue(selector, { errorMessage: `Tram-One: selector should be defined as a CSS Selector or DOM Node, recieved ${selector}` })
+	ensureFunction(component, { errorMessage: `Tram-One: component should be a function, recieved ${typeof selector}, ${selector}` })
+
+	// if the selector is a string, try to find the element,
+	// otherwise it's probably DOM that we should write directly to
+	const target = (typeof selector) === 'string' ? document.querySelector(selector) : selector
+	if (target === null) {
```

Comment:
do you have to be explicit rather than !target?

Reply:
`document.querySelector` specifically returns `null` when an element can't be found... This doesn't do all the type checking I would like, but casting that value to a boolean isn't really required.